### DISCUSSION TITLE
Allow use of Decimal type in enums

### DISCUSSION THREAD

**Discussion by john-h-k** [dotnet/csharplang#2080] [Language Ideas]
Allow use of `Decimal / decimal` type in enums, as such:
```
public enum MyEnum : decimal
{ }
```

The primary motivation for this is that it would allow easy implementation of `FlagsAttribute` style enum to have 128 indepedently XOR-able flags, as opposed to the current 64 bit limit (`Int64 / long`)

**Comment by tannergooding** [comment]
This would require a runtime change in order for it to be a valid `enum` type (and meet the `System.Enum` constraint).

**Comment by YairHalberstadt** [comment]
If you genuinely need more than 64 different values for a flags enum, perhaps it would be ideal to try and break it up into 2 flags?

**Comment by fanoI** [comment]
Probably will be better to have a proper (U)Int128 type with full language support (ability to declare constant literals, to declare them in enum's and so on...).

**Comment by YairHalberstadt** [comment]
@fanol 
That would be useful for creating genuinely unique hashes. With longs you can't avoid the chance of a conflict even with a perfect hash, but with 128 bits, the chance is so small it's practically irrelevant (so long as your hashing algorithm is actually perfect, which is the big if)

**Comment by YairHalberstadt** [comment]
Of course though, writes and reads would not be atomic

**Comment by yaakov-h** [comment]
@YairHalberstadt we'll just have to build 128-bit processors then...

**Comment by YairHalberstadt** [comment]
@yaakov-h 
Sounds like a plan
